### ISSUE TITLE
Looping "Error: Client not initialized" errors when on Cursor Settings > General

### ISSUE THREAD

**Issue by obfuscode** [getcursor/cursor#1958]
## Bug
When in the `Cursor Settings > General` page, the `Extension Host` throws an infinite loop of `Error: Client not initialized` errors.

## How to Reproduce
- Open new Cursor window with or without a workspace
- Cmd/Ctrl-Shift-P, select "Developer: Reload With Extensions Disabled"
- Open Terminal (`Ctrl-\``)
- Switch to `Output` tab
- Select `Extension Host` from Output dropdown

## Impact
**Is Fatal:** No
**Is Breaking:** No
**Priority:** Very Low

## System Information
**Version Info**
```
Version: 0.41.2
VSCode Version: 1.91.1
Commit: 297991d169fa3a11975c5aa95adc69be73250c70
Date: 2024-09-21T04:18:39.370Z (1 day ago)
Electron: 29.4.0
ElectronBuildId: undefined
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Darwin x64 22.6.0
```

**Extensions:** Disabled

## Screenshot of Issue
_Note: Screenshot is with Extensions enabled but the output is identical._
![Cursor-ClientNotInitializedErrors](https://github.com/user-attachments/assets/75515fc7-d7e6-47b8-aa5f-bb533cc292cc)

**Comment by nitneuk** [comment]
@CyrisXD I encountered the same issue, and just saw this warning here https://code.visualstudio.com/api/get-started/your-first-extension
> If you aren't able to see the Hello World command in the debug window, check the package.json file and make sure that engines.vscode version is compatible with the installed version of VS Code.

Setting : 
```
"engines": {
    "vscode": "^1.91.1"
  },
```
in `package.json` to match with vscode cursor's version seems fix it.

**Comment by aayush-kapoor** [comment]
Yeah same for me, trying to develop an extension and keep encountering this 'Client not initialized' error
